ci/scripts/release.sh

```diff
@@ -63,8 +63,8 @@ if [[ -n "${BUILDKITE_TAG}" ]]; then
 fi
 
 echo "--- Build risingwave release binary"
-cargo build -p risingwave_cmd_all --features "rw-static-link" --profile release
-cargo build --bin risectl --features "rw-static-link" --profile release
+cargo build -p risingwave_cmd_all --features rw-static-link --features private-link --profile release
```

Comment:
Please clarify how we handle the `private-link` flag in different environment/setups, such as:
1. ./risedev d
2. docker image
3. single-node mode, etc.

What I concern is that this flag may introduce inconsistent between different setups and we don't have CI to cover the functionality test yet. And we should not trade functionality for faster compile time.

Reply:
https://github.com/risingwavelabs/risingwave/pull/15379 I have another solution which doesn't introduce feature flag.

---

Some thoughts about feature flag:
I admit that feature flag is not easy to use, but since @wangrunji0408 already introduced a feature flag in https://github.com/risingwavelabs/risingwave/pull/15328, adding more features don't make the situation worse. IMO 1 feature is no much difference from N features.. (This is also why I raised this PR at this time)

It's not that hard to get it correct for production (just `release.sh` and `Dockerfile`), and we just need to get it correct once. So I would say we are not "trading functionality" at all. The benefit of faster compile time is too HUGE to ignore..